pkg/operator/controller/ingress/nodeport_service.go

```diff
@@ -24,13 +24,26 @@ import (
 // indicating whether the NodePort service exists, the current NodePort service
 // if it does exist, and an error value.
 func (r *reconciler) ensureNodePortService(ic *operatorv1.IngressController, deploymentRef metav1.OwnerReference) (bool, *corev1.Service, error) {
-	wantService, desired := desiredNodePortService(ic, deploymentRef)
-
 	haveService, current, err := r.currentNodePortService(ic)
 	if err != nil {
 		return false, nil, err
 	}
 
+	// For compatibility, omit the "metrics" port iff the service already
+	// exists and doesn't have a "metrics" port.
```

Comment:
Could you add a comment here mentioning that this logic allows a user to remove the metrics port from the node port service (as discussed offline) ?

Reply:
Added.  Thanks!